games/static/games/proto.js

```diff
@@ -207,83 +238,92 @@ class Board {
 
     moveAliens() {
         this.direction = this.determineDirection()
+        this.alienLocations = this.calculateAlienLocations()
         for (let i = 0; i < this.squares.length - 1; i++) {
-            if(this.squares[i].hasAlien()) {
+            if (this.squares[i].hasAlien()) {
                 let c = this.squares[i].pickCharacterUp()
                 this.aliens.push(c)
             }
         }
-        this.updateAlienLocations()
-        for (let i = 0; i < this.squares.length - 1; i++) {
+        
+        for (let i = 0; i < this.squares.length; i++) {
             if (this.alienLocations.includes(i)) {
-                let c = this.aliens.shift()          
+                let c = this.aliens.shift()
                 this.squares[i].setCharacter(c)
             }
         }
     }
 
-    updateAlienLocations() {
-        return this.alienLocations = this.alienLocations.map(location =>location + this.direction)
-    }
 
-    determineDirection() { 
-        let direction = this.direction    
-        if (this.aliensHitBorder()) {            
-            return this.directionDown()
-        } 
-        if (this.aliensMovingDown()) {
-            if (this.aliensAtLeftBorder()){
-                return DIRECTION_RIGHT
-            }  
-            return DIRECTION_LEFT                        
-        } 
-        return direction        
+    calculateAlienLocations() {
+        return this.squares
+            .filter(square => square.hasAlien())
+            .map(square => square.index + this.direction)
     }
 
-    directionDown() {
-        return this.width
+    determineDirection() {
+        let nextMove = calcNextAlienMove(this.previousDirection, this.counter, 5, this.directionDown())
+        this.direction = nextMove.direction
+        this.previousDirection = nextMove.previousDirection
+        this.counter = nextMove.counter
+        return nextMove.direction
     }
 
-    aliensHitBorder() {
-        return (this.aliensAtLeftBorder() && this.aliensMovingLeft()) || (this.aliensAtRightBorder() && this.aliensMovingRight())
+    changeDirection() {
```

Comment:
changePreviousDirection ?

Reply:
Deleted the whole thing as it was not necessary with the new movement pattern